Sources/PostgresNIO/New/Connection State Machine/ExtendedQueryStateMachine.swift

```diff
@@ -71,6 +75,44 @@ struct ExtendedQueryStateMachine {
             }
         }
     }
+
+    mutating func cancel() -> Action {
+        switch self.state {
+        case .initialized:
+            preconditionFailure("Start must be called immediatly after the query was created")
+
+        case .parseDescribeBindExecuteSyncSent(let queryContext),
+             .parseCompleteReceived(let queryContext),
+             .parameterDescriptionReceived(let queryContext),
+             .rowDescriptionReceived(let queryContext, _),
+             .noDataMessageReceived(let queryContext),
+             .bindCompleteReceived(let queryContext):
+            guard !self.isCancelled else {
+                return .wait
+            }
+
+            self.isCancelled = true
```

Comment:
If you're `guard`ing that `self.isCancelled` is true, you don't need to then set it `true` immediately afterward.

Reply:
Not sure I or you misread the code here: I think, I'm guarding that `isCancelled` is `false`, to then set it to `true`.

Reply:
Whoops, that was my misread! And even more embarrassing 'cause I did the same thing a couple lines down and then caught myself on that one 😅